### PR TITLE
Aadt generic

### PR DESCRIPTION
**Pull request by rdaly525**
Depends on https://github.com/leonardt/hwtypes/pull/94 and hwtypes:new-sum

### CODE REVIEW

File: tests/test_aadt_mapping.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+form_SBV = SBV[num_forms]
+binding_SBV = SBV[max_bindings]
+
+form_var = form_SBV()
+binding_var = binding_SBV()
+
+#arbitrary precision accuract log2 function
+#math.log is not accurate for large powers of 2
+def log2(x):
```
Comment (cdonovick) [inline tests/test_aadt_mapping.py]: is x here a positive python int? If so `x.bit_length()-1` works (`-1` to strip two-complement 0).

File: tests/test_aadt_mapping.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    assert x & (x-1) == 0
+    p=0
+    while 2**p <=x:
+        p+=1
+    p -= 1
+    assert 2**(p) == x
+    return p
+
+def or_reduce(expr_list : tp.List):
+    return reduce(lambda a, b: a|b, expr_list)
```
Comment (cdonovick) [inline tests/test_aadt_mapping.py]:
```
import operator
from functools import partial
or_reduce = partial(operator.or_, reduce)
```

**Comment by rdaly525** [comment]
@cdonovick or @leonardt, could you publish hwtypes? I need changes from https://github.com/leonardt/hwtypes/pull/94 which is not in the latest release.

**Comment by rdaly525** [comment]
@cdonovick, This is not passing travis because the SMT method either takes 1 second or >20 minutes. Could you review the SMT call made in tests/test_aadt_mapping.py to see if there is anything that can be done to decrease this uncertainty and improve performance?

**Comment by rdaly525** [comment]
closing in preference for #110

**Pull request closed without merging**